src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -978,16 +977,304 @@ void CmdSketcherViewSection::activated(int iMsg)
+
+    auto* sizeLabel = new QLabel(QString::fromLatin1("Size"));
+    auto* gridSizeBox = new Gui::QuantitySpinBox();
+    gridSizeBox->setProperty("unit", QVariant(QString::fromLatin1("mm")));
+    gridSizeBox->setObjectName(QString::fromLatin1("gridSize"));
+    gridSizeBox->setToolTip(QApplication::translate("Sketcher_GridSize", "Distance between two subsequent grid lines"));
+    //Is General/GridSize correct? From taskSketcherGeneral it seems yes.
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/General/GridSize");
+    gridSizeBox->setValue(hGrp->GetFloat("GridSize", 10.0));
+    gridSizeBox->setSingleStep(1.0);
+    gridSizeBox->setMaximum(99999999.0);
+    gridSizeBox->setMinimum(0.001);
+    gridSizeBox->setDecimals(3);
+
+    QObject::connect(gridSizeBox, static_cast<void(Gui::QuantitySpinBox::*)(double)>(&Gui::QuantitySpinBox::valueChanged),
+        [=](double val) {
+            auto* sketchView = dynamic_cast<SketcherGui::ViewProviderSketch*>(getActiveGuiDocument()->getInEdit());
+            assert(sketchView);
+
+            sketchView->GridSize.setValue(val);
+    });
+
+    QWidget* gridSizeW = new QWidget;
+    QHBoxLayout* layout = new QHBoxLayout(gridSizeW);
+    layout->addWidget(sizeLabel);
+    layout->addWidget(gridSizeBox);
+    QWidgetAction* a3 = new QWidgetAction(pcAction);
+    a3->setDefaultWidget(gridSizeW);
+    pcAction->addAction(a3);
+
+    a0->setCheckable(true);
+    a1->setCheckable(true);
+    a2->setCheckable(true);
+
+    _pcAction = pcAction;
+    languageChange();
+
+    return pcAction;
+}
+
+void CmdSketcherCompGrid::languageChange()
+{
+    Command::languageChange();
+
+    if (!_pcAction)
+        return;
+    Gui::ActionGroup* pcAction = qobject_cast<Gui::ActionGroup*>(_pcAction);
+    QList<QAction*> a = pcAction->actions();
+
+    QAction* a0 = a[0];
+    a0->setText(QApplication::translate("CmdSketcherGridToggle", "Activate Grid"));
+    a0->setToolTip(QApplication::translate("CmdSketcherGridToggle", "Activate Grid"));
+    a0->setStatusTip(a0->toolTip());
+    QAction* a1 = a[1];
+    a1->setText(QApplication::translate("CmdSketcherGridSnap", "Grid Snap"));
+    a1->setToolTip(QApplication::translate("CmdSketcherGridSnap", "New points will snap to the nearest grid line.\nPoints must be set closer than a fifth of the grid size to a grid line to snap."));
+    a1->setStatusTip(a1->toolTip());
+    QAction* a2 = a[2];
+    a2->setText(QApplication::translate("CmdSketcherGridAuto", "Grid Auto Resize"));
+    a2->setToolTip(QApplication::translate("CmdSketcherGridAuto", "Resize grid automatically depending on zoom."));
+    a2->setStatusTip(a2->toolTip());
+}
+
+bool CmdSketcherCompGrid::isActive()
+{
+    //Initialisation of the commands to the sketch specific settings
+    auto* sketchView = dynamic_cast<SketcherGui::ViewProviderSketch*>(getActiveGuiDocument()->getInEdit());

```

Comment:
This line causes a crash when there is no document because getActiveGuiDocument() returns null.